qiskit_nature/transformers/base_transformer.py

```diff
@@ -13,24 +13,27 @@
 """The Base Operator Transformer interface."""
 
 from abc import ABC, abstractmethod
+from typing import Union
 
+from qiskit_nature import WatsonHamiltonian
 from qiskit_nature.drivers import QMolecule
 
 
 class BaseTransformer(ABC):
-    """The interface for implementing methods which map from one `QMolecule` to another. These
-    methods may or may not affect the size of the Hilbert space.
+    """The interface for implementing methods which map from one `QMolecule` or
+    'WatsonHamiltonian' to another. These methods may or may not affect the size of the Hilbert
+    space.
     """
 
     @abstractmethod
-    def transform(self, q_molecule: QMolecule) -> QMolecule:
-        """Transforms one `QMolecule` into another one. This may or may not affect the size of the
-        Hilbert space.
+    def transform(self, molecule_data: Union[QMolecule, WatsonHamiltonian]):
```

Comment:
Will we ever have transformers that would work on either type. Or more like my comment above would it be better to define A FermionicTransform taking QMolecule and a BosonicTransform taking the WatsonHamiltonian

Reply:
Not sure about it... @mrossinek ? If not, I agree with you.

Reply:
I don't think we will ever have transformers which support both types. Maybe you can also just specify `Any` as the type here and then subclass to a more specific type in those subclasses.

I would even argue that, since we do not even have any transformer for `WatsonHamiltonian` we simply leave that out for this release..